### PR TITLE
F dev branch

### PR DESCRIPTION
**Pull request by felixdasta**
Implemented registration and email validation on the backend. Renamed the trackpack folder to "server" to avoid confusion with the folder named client/trackpack. 

run.py was moved from root to the server folder, established new routings in routes/routes.py and added some new config in __init__.py. Navbar sandwich was implemented.

### CODE REVIEW

File: server/api/dao/user.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @staticmethod
+    def getAllUsers():
+        return User().query.all()
+
+    @staticmethod
+    def getUserById(uid):
+        return User().query.filter_by(user_id=uid).first()
+
+    @staticmethod
+    def getUserByEmail(uemail):
```
Comment (chris810) [inline server/api/dao/user.py:34]: maybe seria useful tambien tener getuserbyusername?

File: server/api/dao/user.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from sqlalchemy.dialects.postgresql import UUID
+from api import db
+import uuid
+
+class User(db.Model):
+    REQUIRED_PARAMETERS = {'email', 'password', 'username'}
+    
+    __tablename__ = 'user'
+    user_id = db.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True, nullable=False)
+    username = db.Column(db.String(30), nullable=False)
```
Comment (chris810) [inline server/api/dao/user.py]: unique?

File: server/api/dao/user.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    email = db.Column(db.String(256), nullable=False)
```
Comment (chris810) [inline server/api/dao/user.py]: unique?

File: server/api/dao/user.py (hunk 1 above)
Comment (felixdasta) [reply server/api/dao/user.py:34]: yes, definitely. lo añado pronto

File: server/api/dao/user.py (hunk 2 above)
Comment (felixdasta) [reply server/api/dao/user.py]: unique lo que implica es que sera unico en la tabla y no se repite, aunque al ser primary key env el parametro no es tan necesario, que se podria remover en el user_id y añadirse en el email y username ya que al esos no ser pk, se puede usar el unique constraint para asegurarnos que sea unique

**Pull request closed without merging**